src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ReferencesPage/ImportedNamespacesValueProvider.cs

```diff
@@ -114,19 +108,19 @@ public override Task<string> OnGetEvaluatedPropertyValueAsync(string propertyNam
                 }
             }
             
-            importsToAdd.Remove(import);
+            importsToAdd.Remove(value);
         }
 
         // Verify we have at least one valid import to add before acquiring a write lock.
-        if (importsToAdd.Any(importToAdd => importToAdd.Length > 0))
+        if (importsToAdd.Any(importToAdd => importToAdd.Key.Length > 0))
         {
             await _projectAccessor.OpenProjectXmlForWriteAsync(_configuredProject.UnconfiguredProject, project =>
             {
-                foreach (string importToAdd in importsToAdd)
+                foreach (KeyValuePair<string, bool> importToAdd in importsToAdd)
```

Comment:
You can use deconstruction here too if you like.

```suggestion
                foreach ((string key, _) in importsToAdd)
```